cdap-ui/app/cdap/components/AbstractWidget/AbstractMultiRowWidget/index.tsx

```diff
@@ -80,26 +81,27 @@ export default class AbstractMultiRowWidget<
           autofocus: null,
         },
         () => {
-          this.addRow(-1, false);
+          if (!this.state.rows || this.state.rows.length === 0) {
+            // adding a new empty row in case values are empty
+            this.addRow(-1, false);
+          }
         }
       );
+    } else {
+      const rows = [];
 
-      return;
```

Comment:
keep this return statement, then the else block is not needed

Reply:
Resolved